src/main/java/com/google/devtools/build/lib/analysis/test/TestActionBuilder.java

```diff
@@ -222,6 +222,14 @@ private TestParams createTestAction(int shards) {
       inputsBuilder.addTransitive(metadataFiles);
       inputsBuilder.addTransitive(
           PrerequisiteArtifacts.nestedSet(ruleContext, ":coverage_support", Mode.DONT_CHECK));
+
+      if (ruleContext.isAttrDefined("$collect_cc_coverage", LABEL)) {
+        Artifact collectCcCoverage =
+                ruleContext.getHostPrerequisiteArtifact("$collect_cc_coverage");
+        inputsBuilder.add(collectCcCoverage);
+        extraTestEnv.put("CC_CODE_COVERAGE_SCRIPT", collectCcCoverage.getExecPathString());
```

Comment:
Maybe abstract CC_CODE_COVERAGE_SCRIPT to a "private static final String" member in this class?